catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/DecimalRangeValidator.java

```diff
@@ -0,0 +1,81 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation.impl.validator;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Set;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.validation.AttributeValidator;
+import ddf.catalog.validation.impl.report.AttributeValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.AttributeValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+import ddf.catalog.validation.violation.ValidationViolation.Severity;
+
+public class DecimalRangeValidator implements AttributeValidator {
+    private final double min;
+
+    private final double max;
+
+    public DecimalRangeValidator(final double min, final double max) {
+        this.min = min;
+        this.max = max;
+    }
+
+    @Override
+    public AttributeValidationReport validate(Attribute attribute) {
+        final AttributeValidationReport report = new AttributeValidationReportImpl();
+
+        if (attribute != null) {
+            final String name = attribute.getName();
+            for (final Serializable value : attribute.getValues()) {
+                final double doubleValue;
+                if (value instanceof Short) {
+                    doubleValue = ((Short) value).doubleValue();
+                } else if (value instanceof Integer) {
+                    doubleValue = ((Integer) value).doubleValue();
+                } else if (value instanceof Long) {
+                    doubleValue = ((Long) value).doubleValue();
+                } else if (value instanceof Float) {
+                    doubleValue = ((Float) value).doubleValue();
+                } else if (value instanceof Double) {
+                    doubleValue = (double) value;
+                } else {
+                    continue;
+                }
+
+                if (!checkRange(doubleValue)) {
+                    final String violationMessage = String.format("%s must be between %f and %f",
+                            name,
+                            min,
+                            max);
+                    final Set<ValidationViolation> violations =
+                            report.getAttributeValidationViolations();
+                    violations.add(new ValidationViolationImpl(Collections.singleton(name),
+                            violationMessage,
+                            Severity.ERROR));
+                    break;
+                }
+            }
+        }
+
+        return report;
+    }
+
+    private boolean checkRange(final double value) {
+        return min <= value && value <= max;
```

Comment:
Probably. Is that something the integrator would specify or would we just have a reasonable default value?